### PR TITLE
[Flang] RFC: Add support for -w option 1/n

### PR DESCRIPTION
**Pull request by kiranchandramohan**
Add support for the -w option to switch OFF all Flang
warnings. This patch only supports switching OFF the
frontend warnings.

TODO : Support for MLIR, LLVM and Driver warnings.
TODO : Support interactions between -w, -pedantic, -Wall

### CODE REVIEW

**Comment by jeffhammond** [comment]
Thanks!  I built it already and will test later today.

**Comment by klausler** [comment]
I am finishing up a patch that ensures that all warning messages are conditional in Semantics.

**Comment by klausler** [comment]
See https://github.com/llvm/llvm-project/pull/90518, which I think would greatly ease implementation of `-w`.

**Comment by jeffhammond** [comment]
This solves the problem that motivated my initial complaint on Slack.

**Comment by jeffhammond** [comment]
Sorry, I bumped the "Ready for Review" button on accident.

**Review by jeffhammond — APPROVED** [review]

**Comment by kiranchandramohan** [comment]
Thanks @jeffhammond. I will wait for @klausler to submit https://github.com/llvm/llvm-project/pull/90518 which will make it easier for the driver and will provide more fine-grained control of warnings.

File: flang/test/Driver/w-option.f90
Comment (banach-space) [inline flang/test/Driver/w-option.f90:1]: [nit] Once you have more than one prefix, `CHECK` becomes noise. At least IMHO 😅  (we all know that these are "check" lines). You could use more descriptive prefixes instead, e.g. `CHECK-PORT` -> `PORTABILITY` (it wasn't obvious to me that `PORT` stood for "portability"), `CHECK` -> `DEFAULT`.

**Review by banach-space — APPROVED** [review]
LGTM, thanks!

File: flang/test/Driver/w-option.f90
Comment (kiranchandramohan) [reply flang/test/Driver/w-option.f90:1]: Done.

**Pull request merged**